### ISSUE TITLE
[prettier-plugin-glsl] Error parsing "varying"

### ISSUE THREAD

**Issue by disambiguator** [NaridaL/glsl-language-toolkit#1]
Hi! I found out about this prettier plugin recently, really cool work!
I noticed that a vertex shader with a declared `varying` does not seem to work. For example:
```glsl
varying vec2 vUv;

    void main() {

    vUv = uv;
    gl_Position = projectionMatrix *
      modelViewMatrix *
      vec4(position,1.0);
    }
```
fails with the error:
```
lib/shaders/fbm.vert[error] lib/shaders/fbm.vert: Error: PARSE ERROR: Expecting: one of these possible Token sequences:
[error]   1. [IDENTIFIER, '(']
[error]   2. [HASH]
[error]   3. [PRAGMA_DIRECTIVE]
[error]   4. ['const', IDENTIFIER, '{']
[error]   5. ['centroid', 'in', IDENTIFIER]
[error]   6. ['centroid', 'out', IDENTIFIER]
[error]   7. ['in', IDENTIFIER, '{']
[error]   8. ['out', IDENTIFIER, '{']
[error]   9. ['uniform', IDENTIFIER, '{']
[error]   10. ['layout', '(', IDENTIFIER]
[error]   11. [INTERPOLATION_QUALIFIER, 'const', IDENTIFIER]
[error]   12. [INTERPOLATION_QUALIFIER, 'centroid', 'in']
[error]   13. [INTERPOLATION_QUALIFIER, 'centroid', 'out']
[error]   14. [INTERPOLATION_QUALIFIER, 'in', IDENTIFIER]
[error]   15. [INTERPOLATION_QUALIFIER, 'out', IDENTIFIER]
[error]   16. [INTERPOLATION_QUALIFIER, 'uniform', IDENTIFIER]
[error]   17. [INTERPOLATION_QUALIFIER, IDENTIFIER, '{']
[error]   18. ['invariant', INTERPOLATION_QUALIFIER, 'const']
[error]   19. ['invariant', INTERPOLATION_QUALIFIER, 'centroid']
[error]   20. ['invariant', INTERPOLATION_QUALIFIER, 'in']
[error]   21. ['invariant', INTERPOLATION_QUALIFIER, 'out']
[error]   22. ['invariant', INTERPOLATION_QUALIFIER, 'uniform']
[error]   23. ['invariant', 'const', IDENTIFIER]
[error]   24. ['invariant', 'centroid', 'in']
[error]   25. ['invariant', 'centroid', 'out']
[error]   26. ['invariant', 'in', IDENTIFIER]
[error]   27. ['invariant', 'out', IDENTIFIER]
[error]   28. ['invariant', 'uniform', IDENTIFIER]
[error]   29. [PRECISION_QUALIFIER]
[error]   30. [BASIC_TYPE]
[error]   31. ['void']
[error]   32. ['struct']
[error]   33. ['const', PRECISION_QUALIFIER]
[error]   34. ['const', BASIC_TYPE]
[error]   35. ['const', 'void']
[error]   36. ['const', 'struct']
[error]   37. ['in', PRECISION_QUALIFIER]
[error]   38. ['in', BASIC_TYPE]
[error]   39. ['in', 'void']
[error]   40. ['in', 'struct']
[error]   41. ['out', PRECISION_QUALIFIER]
[error]   42. ['out', BASIC_TYPE]
[error]   43. ['out', 'void']
[error]   44. ['out', 'struct']
[error]   45. ['uniform', PRECISION_QUALIFIER]
[error]   46. ['uniform', BASIC_TYPE]
[error]   47. ['uniform', 'void']
[error]   48. ['uniform', 'struct']
[error]   49. [INTERPOLATION_QUALIFIER, PRECISION_QUALIFIER]
[error]   50. [INTERPOLATION_QUALIFIER, BASIC_TYPE]
[error]   51. [INTERPOLATION_QUALIFIER, 'void']
[error]   52. [INTERPOLATION_QUALIFIER, 'struct']
[error]   53. [IDENTIFIER, '[']
[error]   54. [IDENTIFIER, IDENTIFIER]
[error]   55. [IDENTIFIER, ';']
[error]   56. ['const', IDENTIFIER, '[']
[error]   57. ['const', IDENTIFIER, IDENTIFIER]
[error]   58. ['const', IDENTIFIER, ';']
[error]   59. ['centroid', 'in', PRECISION_QUALIFIER]
[error]   60. ['centroid', 'in', BASIC_TYPE]
[error]   61. ['centroid', 'in', 'void']
[error]   62. ['centroid', 'in', 'struct']
[error]   63. ['centroid', 'in', IDENTIFIER]
[error]   64. ['centroid', 'out', PRECISION_QUALIFIER]
[error]   65. ['centroid', 'out', BASIC_TYPE]
[error]   66. ['centroid', 'out', 'void']
[error]   67. ['centroid', 'out', 'struct']
[error]   68. ['centroid', 'out', IDENTIFIER]
[error]   69. ['in', IDENTIFIER, '[']
[error]   70. ['in', IDENTIFIER, IDENTIFIER]
[error]   71. ['in', IDENTIFIER, ';']
[error]   72. ['out', IDENTIFIER, '[']
[error]   73. ['out', IDENTIFIER, IDENTIFIER]
[error]   74. ['out', IDENTIFIER, ';']
[error]   75. ['uniform', IDENTIFIER, '[']
[error]   76. ['uniform', IDENTIFIER, IDENTIFIER]
[error]   77. ['uniform', IDENTIFIER, ';']
[error]   78. ['layout', '(', IDENTIFIER]
[error]   79. [INTERPOLATION_QUALIFIER, 'const', PRECISION_QUALIFIER]
[error]   80. [INTERPOLATION_QUALIFIER, 'const', BASIC_TYPE]
[error]   81. [INTERPOLATION_QUALIFIER, 'const', 'void']
[error]   82. [INTERPOLATION_QUALIFIER, 'const', 'struct']
[error]   83. [INTERPOLATION_QUALIFIER, 'const', IDENTIFIER]
[error]   84. [INTERPOLATION_QUALIFIER, 'centroid', 'in']
[error]   85. [INTERPOLATION_QUALIFIER, 'centroid', 'out']
[error]   86. [INTERPOLATION_QUALIFIER, 'in', PRECISION_QUALIFIER]
[error]   87. [INTERPOLATION_QUALIFIER, 'in', BASIC_TYPE]
[error]   88. [INTERPOLATION_QUALIFIER, 'in', 'void']
[error]   89. [INTERPOLATION_QUALIFIER, 'in', 'struct']
[error]   90. [INTERPOLATION_QUALIFIER, 'in', IDENTIFIER]
[error]   91. [INTERPOLATION_QUALIFIER, 'out', PRECISION_QUALIFIER]
[error]   92. [INTERPOLATION_QUALIFIER, 'out', BASIC_TYPE]
[error]   93. [INTERPOLATION_QUALIFIER, 'out', 'void']
[error]   94. [INTERPOLATION_QUALIFIER, 'out', 'struct']
[error]   95. [INTERPOLATION_QUALIFIER, 'out', IDENTIFIER]
[error]   96. [INTERPOLATION_QUALIFIER, 'uniform', PRECISION_QUALIFIER]
[error]   97. [INTERPOLATION_QUALIFIER, 'uniform', BASIC_TYPE]
[error]   98. [INTERPOLATION_QUALIFIER, 'uniform', 'void']
[error]   99. [INTERPOLATION_QUALIFIER, 'uniform', 'struct']
[error]   100. [INTERPOLATION_QUALIFIER, 'uniform', IDENTIFIER]
[error]   101. [INTERPOLATION_QUALIFIER, IDENTIFIER, '[']
[error]   102. [INTERPOLATION_QUALIFIER, IDENTIFIER, IDENTIFIER]
[error]   103. [INTERPOLATION_QUALIFIER, IDENTIFIER, ';']
[error]   104. ['invariant', INTERPOLATION_QUALIFIER, 'const']
[error]   105. ['invariant', INTERPOLATION_QUALIFIER, 'centroid']
[error]   106. ['invariant', INTERPOLATION_QUALIFIER, 'in']
[error]   107. ['invariant', INTERPOLATION_QUALIFIER, 'out']
[error]   108. ['invariant', INTERPOLATION_QUALIFIER, 'uniform']
[error]   109. ['invariant', 'const', PRECISION_QUALIFIER]
[error]   110. ['invariant', 'const', BASIC_TYPE]
[error]   111. ['invariant', 'const', 'void']
[error]   112. ['invariant', 'const', 'struct']
[error]   113. ['invariant', 'const', IDENTIFIER]
[error]   114. ['invariant', 'centroid', 'in']
[error]   115. ['invariant', 'centroid', 'out']
[error]   116. ['invariant', 'in', PRECISION_QUALIFIER]
[error]   117. ['invariant', 'in', BASIC_TYPE]
[error]   118. ['invariant', 'in', 'void']
[error]   119. ['invariant', 'in', 'struct']
[error]   120. ['invariant', 'in', IDENTIFIER]
[error]   121. ['invariant', 'out', PRECISION_QUALIFIER]
[error]   122. ['invariant', 'out', BASIC_TYPE]
[error]   123. ['invariant', 'out', 'void']
[error]   124. ['invariant', 'out', 'struct']
[error]   125. ['invariant', 'out', IDENTIFIER]
[error]   126. ['invariant', 'uniform', PRECISION_QUALIFIER]
[error]   127. ['invariant', 'uniform', BASIC_TYPE]
[error]   128. ['invariant', 'uniform', 'void']
[error]   129. ['invariant', 'uniform', 'struct']
[error]   130. ['invariant', 'uniform', IDENTIFIER]
[error]   131. ['precision']
[error] but found: 'varying'
[error] translationUnit > externalDeclaration
[error] ./src/compiler.lambda:1:0
[error] 
[error]     at checkParsingErrors (/Users/paras/workspace/floating_points/node_modules/prettier-plugin-glsl/lib/prettier-plugin.cjs.js:2461:15)
[error]     at parseInput (/Users/paras/workspace/floating_points/node_modules/prettier-plugin-glsl/lib/prettier-plugin.cjs.js:2493:5)
[error]     at Object.parse (/Users/paras/workspace/floating_points/node_modules/prettier-plugin-glsl/lib/prettier-plugin.cjs.js:2577:37)
[error]     at Object.parse (/Users/paras/workspace/floating_points/node_modules/prettier/index.js:7334:23)
[error]     at coreFormat (/Users/paras/workspace/floating_points/node_modules/prettier/index.js:8645:18)
[error]     at formatWithCursor2 (/Users/paras/workspace/floating_points/node_modules/prettier/index.js:8837:18)
[error]     at Object.formatWithCursor (/Users/paras/workspace/floating_points/node_modules/prettier/index.js:37229:12)
[error]     at format (/Users/paras/workspace/floating_points/node_modules/prettier/cli.js:14748:24)
[error]     at formatFiles2 (/Users/paras/workspace/floating_points/node_modules/prettier/cli.js:14860:22)
[error]     at async main (/Users/paras/workspace/floating_points/node_modules/prettier/cli.js:15077:5)
```

But if I comment out the first line as...
```
// varying vec2 vUv;

    void main() {

    vUv = uv;
    gl_Position = projectionMatrix *
      modelViewMatrix *
      vec4(position,1.0);
    }

```

I correctly get:
```
// varying vec2 vUv;

void main() {
  vUv = uv;
  gl_Position = projectionMatrix * modelViewMatrix * vec4(position, 1.0);
}
```

**Comment by NaridaL** [comment]
Thanks!

Yeah, currently it only targets OpenGL ES 3.00, which doesn't support the ES 1.00 keyword varying. It would probably make sense to support both, yeah.

**Comment by NaridaL** [comment]
I published v0.0.5, give it a go.

**Comment by disambiguator** [comment]
Yep that fixed it. Thank you!
